src/main/java/com/pusher/client/PusherOptions.java

```diff
@@ -196,6 +198,27 @@ public String buildUrl(final String apiKey) {
                 : wsPort, apiKey, URI_SUFFIX);
     }
 
+    /**
+     *
+     * The default value is Proxy.NO_PROXY.
+     *
+     * @param proxy
+     *            Specify a proxy, e.g. <code>options.setProxy( new Proxy( Proxy.Type.HTTP, new InetSocketAddress( "proxyaddress", 80 ) ) )</code>;
+     * @return this, for chaining
+     */
+    public PusherOptions setProxy(Proxy proxy){
+        this.proxy = proxy;
+        return this;
+    }
+
+    public Proxy getProxy() {
+        if (this.proxy == null) {
```

Comment:
UNLESS you're providing a matching pair of get/set which apply the same transformation in both directions. But if you call `a.setX(x)`, then `a.getX() == x` should always be true.